### PR TITLE
Remove duplicate dex_router state

### PR DESCRIPTION
**Pull request by lubkoll**

### CODE REVIEW

**Review by arhamchordia — APPROVED** [review]
Went through it. Looks good. 
Can merge once there is another review.

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs [hunk 1]
```diff
@@ -57,7 +56,7 @@ pub(crate) fn execute_any_deposit(
     }
 
     let twap_price = get_twap_price(deps.storage, &deps.querier, &env, 24u64)?;
-    let (token_in, out_denom, remainder, price) = if !deposit_info.base_refund.amount.is_zero() {
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs:60]: nit: why does a pr about removing dex router change remainder here?

Comment (LaurensKubat) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs:60]: Also not sure why we are changing this

**Review by LaurensKubat — DISMISSED** [review]
lgtm

**Review by LaurensKubat — COMMENTED** [review]
Not sure about why we have certain changes to deposit.rs other than that lgtm and ready to merge once explained or sorted

**Comment by lubkoll** [comment]
> Not sure about why we have certain changes to deposit.rs other than that lgtm and ready to merge once explained or sorted

:D yes, sry for the confusion, probably should have left a comment of all the things I changed to make this actually work as that was puzzling for me too. So here is roughly why the unexpected changes are here.
* in the test-tube tests we were mainly assuming that the dex_router is not set and were doing direct swaps against osmosis
* when handling the response from AnyDeposit, then this was also only tested directly against osmosis, and incorrectly assuming that we get a direct response from osmosis to read the received tokens (i.e. we did a cast of the result to a response from osmosis). This of course failed when using the dex_router. So I removed that and instead just used the balances from the contract to establish a new position. As a consequence all the data stored in CURRENT_SWAP_ANY_DEPOSIT were not used anymore, except the recipient, so  I removed these too.
* as by now you may have guessed all the changes of the values in the tests are also due to swaps through the dex_router instead of direct swaps against osmosis

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs (hunk 1 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs:60]: see long explanation below

Comment (LaurensKubat) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs:60]: Makes sense now, to add to the confusion github's view cut this weirdly to make me think this was part of exact_deposit 😓

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Review by magiodev — APPROVED** [review]

**Review by magiodev — CHANGES_REQUESTED** [review]
dmed

**Review by ajansari95 — APPROVED** [review]
lgtm

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Review by magiodev — APPROVED** [review]
looks good now after the adjustments

**Pull request merged**